hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -217,6 +218,7 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tablePath: String,
                 .withStart(file.start)
                 .withLength(baseFileLength)
                 .withShouldUseRecordPosition(shouldUseRecordPosition)
+                .withEnableOptimizedLogBlockScan(props.getBoolean(HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.key(), HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.defaultValue().toBoolean))
```

Comment:
if the option key is already in the props, we can add a fallback logic in the FG reader builder:

```java
if (withEnableOptimizedLogBlockScan == null) {
  withEnableOptimizedLogBlockScan = props.getBoolean(HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.key(), HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.defaultValue().toBoolean)
}
```

Reply:
if the value is false and found `props.getBoolean(HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.key(), HoodieReaderConfig.ENABLE_OPTIMIZED_LOG_BLOCKS_SCAN.defaultValue().toBoolean)` as true, override it , does it make sense?

Reply:
Was wondering if you can provide a little more clarity on what you mean by `if the value is false`, could you maybe provide a code pointer?

Reply:
Synced with @danny0405 offline over slack, pasting my discussion here

<img width="1823" height="921" alt="Screenshot 2025-07-24 at 8 43 19 AM" src="https://github.com/user-attachments/assets/72e90718-39c8-4a50-998b-57be2d78fc5c" />